apps/easypid/src/features/didcomm/DidCommNotificationScreen.tsx

```diff
@@ -113,50 +120,76 @@ export function DidCommNotificationScreen() {
     }
   }, [params.invitation, params.invitationUrl, hasHandledNotificationLoading, agent, isDevelopmentModeEnabled])
 
-  // Both flows have the same entry point, so we re-use the same loading request slide
-  // This way we avoid a double loading screen when the respective flow is entered
-  const steps: SlideStep[] = [
-    {
-      step: 'loading-request',
-      progress: 33,
-      screen: <LoadingRequestSlide key="loading-request" isLoading={!notification} isError={!!errorReason} />,
-    },
-  ]
-
-  const credentialSlides = useDidCommCredentialNotificationSlides({
-    credentialExchangeId: notification?.id as string,
-    onCancel,
-    onComplete,
-  })
-  const presentationSlides = useDidCommPresentationNotificationSlides({
-    proofExchangeId: notification?.id as string,
-    onCancel,
-    onComplete,
-  })
-
-  // Add the appropriate slides based on notification type
-  if (notification) {
-    if (notification.type === 'credentialExchange') {
-      steps.push(...credentialSlides)
-    } else if (notification.type === 'proofExchange') {
-      steps.push(...presentationSlides)
+  // Delay the navigation to hide the fact we're loading in the new slides based on the flow type
+  useEffect(() => {
+    if (flow) {
+      setTimeout(() => {
+        setReadyToNavigate(true)
+      }, 250)
     }
+  }, [flow])
+
+  if (flow?.type === 'connect' && readyToNavigate) {
+    return <ConnectionSlides name={display.connection.name} onCancel={onCancel} onComplete={onComplete} />
+  }
+
+  if ((flow?.type === 'issue' && readyToNavigate) || params.credentialExchangeId) {
+    const isExistingExchange = !!params.credentialExchangeId
+    const id = isExistingExchange ? params.credentialExchangeId : flow?.id
+
+    return (
```

Comment:
Eventually it would be nice that if you're navigating here from an existing notification we still show the party you're interacting with? But that's definitely for a later iteration

Reply:
Maybe we can show that in the notification list item?